src/libstd/sys/unix/ext/fs.rs

```diff
@@ -20,6 +20,44 @@ use sys;
 use sys_common::{FromInner, AsInner, AsInnerMut};
 use sys::platform::fs::MetadataExt as UnixMetadataExt;
 
+/// Unix-specific extensions to `File`
+#[unstable(feature = "file_offset", issue = "35918")]
+pub trait FileExt {
+    /// Reads a number of bytes starting from a given offset.
+    ///
+    /// The offset is relative to the start of the file and thus independent
+    /// from the current cursor.
+    ///
+    /// The current file cursor is not affected by this function.
+    ///
+    /// Note that similar to `File::read`, it is not an error to return with a
+    /// short read.
```

Comment:
Could these docs also clarify what happens if you read beyond the end of a file? E.g. if a file is 10 bytes long what do these return:

* `read_at(buf, 10)`
* `read_at(buf, 100)`